### DISCUSSION TITLE
How do we set "variables" in tests

### DISCUSSION THREAD

**Discussion by mganchas** [mobile-dev-inc/Maestro#225] [Q&A] [answered]
Hey! Great work on Maestro 👌 Does anyone know how to create variables in our tests?
For example:

```
appId: com.company.app
---

- variables:
   loginText: "Login to your account"

- assertVisible: $loginText
- tapOn: $loginText

```

Thanks!

**Comment by kaling852** [comment]
I follow this doc and and I pass the variables from the command line. 
https://maestro.mobile.dev/getting-started/maestro-cli#pass-variables-to-tests

**Comment by nephix** [comment]
I'm also interested in doing this.  My use case is that I want to do a user registration flow:

1. User signs up via email/password
2. Verification token gets created on server
3. Somehow fetch this token during the test via REST API
4. Open link via maestro like https://example.org/verify/token
5. Opens app and verifies user's email

**Comment by axelniklasson** [comment, marked as answer]
Hi folks! This is indeed available in maestro and you can find the docs regarding passing variables to your tests [here](https://maestro.mobile.dev/advanced/parameters-and-constants). Thanks!
